api/v1beta1/topicpermission_types.go

```diff
@@ -0,0 +1,83 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// TopicPermissionSpec defines the desired state of TopicPermission
+type TopicPermissionSpec struct {
```

Comment:
We noticed that TopicPermissionSpec struct is identical to PermissionSpec of the Permission type. It would be possible to reuse it or maybe is good to have them separate for future extensions?

Reply:
Structure of their the permission configurations are different: https://github.com/rabbitmq/messaging-topology-operator/blob/topic-permission-191/api/v1beta1/permission_types.go#L30-L37

User permissions needs `configrue` `read` and `write`, whereas topic permission is name of the exchange, `read` and `write`.